jscomp/others/js_re.ml

```diff
@@ -40,9 +40,26 @@
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU Lesser General Public License for more details.
- * 
+ *
  * You should have received a copy of the GNU Lesser General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. *)
 
 type t
+type result = string array
+(* `result` also has `index` and `input` properties, but don't know how to represent that *)
```

Comment:
can you type `result` as abstract type and type it as an array like object instead?

```ocaml
type result 
external matches : result -> string array ..
external index : result -> ...
```